events/views.py

```diff
@@ -137,4 +137,35 @@ def edit_event(request, event_id):
     context = {
         'form': form, 'event': event
     }
+    return render(request, template, context)
+
+
+    @login_required
+def delete_event(request, event_id):
+    """
+    Allow admin users to edit events on the site.
+    """
+    if not request.user.is_superuser:
+        messages.error(request, "Access Denied: Invalid Credentials")
+        return redirect("index")
+
+    event = get_object_or_404(Event, id=event_id)
+
+    if request.method == "POST":
```

Comment:
This can be shortened since there is no `if` statement for the `else` (or, it's not aligned properly on the parent.

Try this?

```python
    if request.method == "POST":
        event.delete()
            messages.success(
                request,
                'Event deleted successfully'
            )
            return redirect('events')

    template = 'events/delete_event.html'
    context = {
        'event': event
    }
    return render(request, template, context)
    ```